api/envoy/config/route/v3/route_components.proto

```diff
@@ -2384,4 +2384,18 @@ message FilterConfig {
   // not support the specified filter, it may ignore the map entry rather
   // than rejecting the config.
   bool is_optional = 2;
+
+  // If true, the filter is disabled in the route or virtual host and the ``config`` field is ignored.
+  //
+  // .. note::
+  //
+  //   This field will take effect when the request arrive and filter chain is created for the request.
+  //   If initial route is selected for the request and a filter is disabled in the initial route, then
+  //   the filter will not be added to the filter chain.
+  //   And if the request is mutated later and re-match to another route, the disabled filter by the
+  //   initial route will not be added back to the filter chain because the filter chain is already
+  //   created and it is too late to change the chain.
+  //
+  //   This field only make sense for the downstream HTTP filters for now.
```

Comment:
Should add a not-implemented-yet annotation or merge this into the implementation PR.

Reply:
The `FilterConfig` message are annotated as `not-implemented`. So, it's unnecessary to add the annotation seperatedly for this field.